diff --git a/node_modules/leaflet/dist/leaflet-src.js b/node_modules/leaflet/dist/leaflet-src.js
index 533f4e5..56edce3 100644
--- a/node_modules/leaflet/dist/leaflet-src.js
+++ b/node_modules/leaflet/dist/leaflet-src.js
@@ -4518,16 +4518,16 @@
 
   	_fireDOMEvent: function (e, type, canvasTargets) {
 
-  		if (e.type === 'click') {
-  			// Fire a synthetic 'preclick' event which propagates up (mainly for closing popups).
-  			// @event preclick: MouseEvent
-  			// Fired before mouse click on the map (sometimes useful when you
-  			// want something to happen on click before any existing click
-  			// handlers start running).
-  			var synth = extend({}, e);
-  			synth.type = 'preclick';
-  			this._fireDOMEvent(synth, synth.type, canvasTargets);
-  		}
+  		// if (e.type === 'click') {
+  		// 	// Fire a synthetic 'preclick' event which propagates up (mainly for closing popups).
+  		// 	// @event preclick: MouseEvent
+  		// 	// Fired before mouse click on the map (sometimes useful when you
+  		// 	// want something to happen on click before any existing click
+  		// 	// handlers start running).
+  		// 	var synth = extend({}, e);
+  		// 	synth.type = 'preclick';
+  		// 	this._fireDOMEvent(synth, synth.type, canvasTargets);
+  		// }
 
   		// Find the layer the event is propagating from and its parents.
   		var targets = this._findEventTargets(e, type);
